assignments/prototypes.js

```diff
@@ -148,4 +148,123 @@ Humanoid.prototype.greet = function() {
+    if(target.healthPoints <= 0){
+      return `${this.name} just smited ${target.name} for ${damage} damage!\n${target.name} has died.`
+    }
+    return `${this.name} just smited ${target.name} for ${damage} damage!\n${target.name} has ${target.healthPoints} health left.`
+  }
+
+  Hero.prototype.heal = function() {
+    let healAmt = Math.floor(Math.random() * 11);
+    this.healthPoints += healAmt;
+    return `${this.name} just self healed ${healAmt} health points and is now at ${this.healthPoints} health.`;
+  }
+
+  function Villain(char){
+    Humanoid.call(this, char)
+    this.reputation = 'Villain';
+  }
+  
+  //Declare this after Constructor function declaration, this way it can inherit the immediate parents prototype functions.
+  Villain.prototype = Object.create(Humanoid.prototype);
+  
+  Villain.prototype.burn = function(target) {
+    let damage = Math.floor(Math.random() * 16);
+    target.healthPoints -= damage;
+    if(target.healthPoints <= 0){
+      return `${this.name} just burned ${target.name} for ${damage} damage!\n${target.name} has died.`
+    }
+    return `${this.name} just burned ${target.name} for ${damage} damage!\n${target.name} has ${target.healthPoints} health left.`
+  }
+
+  Villain.prototype.sacrifice = function(target) {
+    let damage = Math.floor(Math.random() * 14);
+    target.healthPoints -= damage;
+    let sacrificeAmt = Math.floor(damage * .5);
+    this.healthPoints -= sacrificeAmt;
+    if(target.healthPoints <= 0){
+      return `${this.name} just sacrificed ${sacrificeAmt} points of their own Health to deal ${damage} to defeat ${target.name}`;
+    } else if(this.healthPoints <= 0) {
+      return `${this.name} just sacrificed ${sacrificeAmt} points of their own Health to deal ${damage} to ${target.name} - but died in the process... `;
+    } else if(this.healthPoints <= 0 && target.healthPoints <= 0){
+      return `${this.name} just sacrificed ${sacrificeAmt} points of their own Health to deal ${damage} to ${target.name}.\nThis attack defeated ${this.name}, as well as ${target.name}`;
+    }
+    return `${this.name} just sacrificed ${sacrificeAmt} points of their own Health to deal ${damage} to ${target.name}\n${target.name} has ${target.healthPoints} health left\n${this.name} has ${this.healthPoints} health left`;
+  }
+
+  const hero = new Hero({
+    createdAt: new Date(),
+    dimensions: {
+      length: 1,
+      width: 2,
+      height: 4,
+    },
+    healthPoints: 50,
+    name: 'Hero',
+    team: 'Forest Kingdom',
+    weapons: [
+      'Bow',
+      'Dagger',
+    ],
+    language: 'Elvish',
+  });
+
+  const villain = new Villain({
+    createdAt: new Date(),
+    dimensions: {
+      length: 1,
+      width: 2,
+      height: 4,
+    },
+    healthPoints: 50,
+    name: 'Villain',
+    team: 'Forest Kingdom',
+    weapons: [
+      'Bow',
+      'Dagger',
+    ],
+    language: 'Elvish',
+  });
+
+  function battleSequence(player1, player2) {
```

Comment:
Great job creating a variety of methods for both players. including a method that affect health points depending on the outcome of a function using math.random.